src/animationLoop.js

```diff
@@ -0,0 +1,117 @@
+function isActiveSubscription(subscription) {
+  return subscription.active;
+}
+
+const prototype = {
+  running: false,
+  shouldStop: false,
+  lastTime: 0,
+  accumulatedTime: 0,
+
+  setOptions: function setOptions(options) {
+    const animationLoop = this;
+    const timeStep = animationLoop.timeStep = options.timeStep;
+    animationLoop.timeScale = options.timeScale;
+    animationLoop.maxSteps = options.maxSteps;
+    animationLoop.getTime = options.getTime;
+    animationLoop.ticker = options.ticker;
+
+    animationLoop.step = function step(subscriber) {
+      if (subscriber.active) {
+        subscriber.value = subscriber.step(subscriber.value, timeStep);
+      }
+    };
+
+    return animationLoop;
+  },
+
+  subscribe: function subscribe(value, step, render) {
+    const animationLoop = this;
+    const subscriber = {
+      value: value,
+      step: step,
+      render: render,
+      active: true,
+    };
+
+    animationLoop.state.push(subscriber);
+
+    return function unsubscribe() {
+      subscriber.active = false;
+    };
+  },
+
+  loop: function loop() {
+    const animationLoop = this;
+    const currentTime = animationLoop.getTime();
+
+    if (animationLoop.shouldStop) {
+      animationLoop.running = animationLoop.shouldStop = false;
+      return;
+    }
+
+    const timeStep = animationLoop.timeStep;
+    const frameTime = currentTime - animationLoop.lastTime;
+
+    animationLoop.lastTime = currentTime;
+    animationLoop.accumulatedTime += frameTime * animationLoop.timeScale;
+
+    if (animationLoop.accumulatedTime > timeStep * animationLoop.maxSteps) {
+      animationLoop.accumulatedTime = 0;
+    }
+
+    while (animationLoop.accumulatedTime > 0) {
```

Comment:
Probably having independent animation loop for each animation.

Or/and deferring calculations